### PR TITLE
Feature/eicnet 2722 flag access check

### PR DESCRIPTION
**Pull request by lramarojaona**
### Improvements

- Moved our custom logic for `isPublished()` and `isArchived()` to a new `Drupal\eic_moderation\ModerationHelper` service
- Refactored the code in theme for testing the access to share

### Tests

- [x] As SA, create a Public group and publish it, make sure you can like and share on social media
- [x] Create a Document, Discussion, Video, Gallery and Event as Draft (unpublished)
- [x] Make you sure you cannot like and share on all of them
- [x] Publish them and make sure you can like and share
- [x] Unpublish the group
- [x] Make sure you cannot like and share the Group, Document, Discussion, Video, Gallery and Event

### CODE REVIEW

**Review by 4hardfire — CHANGES_REQUESTED** [review]
Code seems fine but some steps didn't pass:

- `As SA, create a Public group and publish it, make sure you can like and share on social media`
When the group is pending I still see the share link. Also, for some reason the like button is never shown even if the group is published

- `Make you sure you cannot like and share on all of them`
After creating a group content, the share button is always shown (even in draft). The like button is ok.

- `Make sure you cannot like and share the Group, Document, Discussion, Video, Gallery and Event`
The like button is never shown in the group header (even when group is published) and the share button is always shown independently of the group status. For the group content, the like button is ok but the share button is always shown independently of the node moderation state or group moderation state.

File: lib/modules/eic_moderation/src/ModerationHelper.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
+<?php
```
Comment (4hardfire) [inline lib/modules/eic_moderation/src/ModerationHelper.php:1]: I would put this file inside eic_moderation/src/Services

**Review by 4hardfire — CHANGES_REQUESTED** [review]

File: lib/modules/eic_moderation/src/ModerationHelper.php (hunk 1 above)
Comment (lramarojaona) [reply lib/modules/eic_moderation/src/ModerationHelper.php:1]: Well, we already have lots of Helper classes that are the src root

**Comment by lramarojaona** [comment]
> Code seems fine but some steps didn't pass:
> 
>     * `As SA, create a Public group and publish it, make sure you can like and share on social media`
>       When the group is pending I still see the share link. Also, for some reason the like button is never shown even if the group is published
> 
>     * `Make you sure you cannot like and share on all of them`
>       After creating a group content, the share button is always shown (even in draft). The like button is ok.
> 
>     * `Make sure you cannot like and share the Group, Document, Discussion, Video, Gallery and Event`
>       The like button is never shown in the group header (even when group is published) and the share button is always shown independently of the group status. For the group content, the like button is ok but the share button is always shown independently of the node moderation state or group moderation state.

@4hardfire , there was some code messing up tieh flags, so the like was always removed.

I fixed the otcher checks.

Can you re-test?

File: lib/themes/eic_community/includes/preprocess/blocks/block--eic-group-header-block.inc [hunk 2]
```diff
@@ -198,12 +196,6 @@ function eic_community_preprocess_eic_group_header_block(array &$variables) {
       }
     }
 
-    // We set the recommend flag in the 2nd position.
-    if ($name === 'recommend_content') {
-      $flags[1] = $flag;
-      continue;
-    }
-
```
Comment (4hardfire) [inline lib/themes/eic_community/includes/preprocess/blocks/block--eic-group-header-block.inc:206]: Is this removal intended? I remember this was previously requested by the client, to have the Recommend flag in the second position.

**Review by 4hardfire — CHANGES_REQUESTED** [review]

**Comment by 4hardfire** [comment]
> > Code seems fine but some steps didn't pass:
> > ```
> > * `As SA, create a Public group and publish it, make sure you can like and share on social media`
> >   When the group is pending I still see the share link. Also, for some reason the like button is never shown even if the group is published
> > 
> > * `Make you sure you cannot like and share on all of them`
> >   After creating a group content, the share button is always shown (even in draft). The like button is ok.
> > 
> > * `Make sure you cannot like and share the Group, Document, Discussion, Video, Gallery and Event`
> >   The like button is never shown in the group header (even when group is published) and the share button is always shown independently of the group status. For the group content, the like button is ok but the share button is always shown independently of the node moderation state or group moderation state.
> > ```
> 
> @4hardfire , there was some code messing up tieh flags, so the like was always removed.
> 
> I fixed the otcher checks.
> 
> Can you re-test?

This is ok!

File: lib/modules/eic_moderation/src/ModerationHelper.php (hunk 1 above)
Comment (4hardfire) [reply lib/modules/eic_moderation/src/ModerationHelper.php:1]: ok

**Pull request merged**